libosrmc/osrmc.cc

```diff
@@ -376,3 +411,98 @@ void osrmc_match_params_add_timestamp(osrmc_match_params_t params, unsigned time
 } catch (const std::exception& e) {
   osrmc_error_from_exception(e, error);
 }
+
+
+/* Trip service */
+osrmc_trip_params_t osrmc_trip_params_construct(osrmc_error_t* error) try {
+  osrm::TripParameters* out = new osrm::TripParameters;
+  return reinterpret_cast<osrmc_trip_params_t>(out);
+} catch (const std::exception& e) {
+  osrmc_error_from_exception(e, error);
+  return nullptr;
+}
+
+void osrmc_trip_params_destruct(osrmc_trip_params_t params) {
+  delete reinterpret_cast<osrm::TripParameters*>(params);
+}
+
+void osrmc_trip_params_add_source(osrmc_trip_params_t params, bool first, osrmc_error_t* error) try {
+  auto* params_typed = reinterpret_cast<osrm::TripParameters*>(params);
+
+  if (first)
+    params_typed->source = osrm::TripParameters::SourceType::First;
```

Comment:
https://github.com/Project-OSRM/osrm-backend/blob/15f0ca8ddaa35c5b4d93c25afa72e81e1fb40c3e/include/engine/api/trip_parameters.hpp#L52-L61

We should support all enum values here; not sure a `bool first` is enough here.